### DISCUSSION TITLE
How can I update my database through stripe webhook? (sveltekit+pocketbase+stripe)

### DISCUSSION THREAD

**Discussion by idea2547** [pocketbase/pocketbase#3910] [Q&A]
This is my switch event type in my webhook code I want to update my subscription status but it didn't work - I successfully pass through the switch case but locals.pb.collection('users').update(locals.user.id, { Subscription }); not work:

switch (event.type) {
		case 'payment_intent.succeeded':
			const paymentIntentSucceeded = event.data.object;
			console.log(paymentIntentSucceeded)
			console.log("Payment Success 101")
			await locals.pb.collection('users').update(locals.user.id, { Subscription });
			// await locals.pb.collection('users').update(locals.user.id, { Token });
			// Then define and call a function to handle the event payment_intent.succeeded
			break;

**Comment by ganigeorgiev** [comment]
There is really not enough details to help you debug it.

Wrap the update call with try/catch and add `console.log(err.response)` to inspect the error and/or check the failed request in the Admin UI > Logs.

Double check your collection API rules and whether the `locals.pb` client is authenticated.

Additionally note that usually payment related states are expected to be performed **only** by an admin, aka. regular users shouldn't be able to change directly their own subscription state. If that's the case either make sure that the appropriate API rules are defined OR move the "private" fields to a new collection that only admins will have access. And then in your stripe webhook handler create an admin-only client similar to https://github.com/pocketbase/pocketbase/issues/3903#issuecomment-1847842771 and send the update request with it.
